docs/src/main/asciidoc/picocli.adoc

```diff
@@ -88,11 +91,71 @@ public class EntryCommand {
     public void setName(String name) {
         this.name = name;
     }
-
 }
 ----
 
-== Command line application with multiple Commands
+=== Packaging
+
+Picocli provides different ways
+
+> to package your application in such a way that end users can invoke it by its command name. https://picocli.info/#_packaging_your_application[Source]
+
+If you want to distribute a single executable file that runs without installing a JDK see <<Native mode support>>. Notice that this executable will be platform specific.
+
+If you want a platform independent solution you can create an executable uber-jar. This uber-jar will contain all dependencies of your app, but still requires an installed JDK to work. You can create an uber-jar via a single property in your `pom.xml`. With the https://github.com/brianm/really-executable-jars-maven-plugin[really-executable-jar-maven-plugin] you can make the uber-jar executable.
+
+[source, xml]
+----
+<properties>
+  <quarkus.package.type>uber-jar</quarkus.package.type> <1>
+  <jar.finalName>${project.name}</jar.finalName> <2>
+</properties>
+
+<!-- make the jar chmod +x style executable -->
+<!-- more options at https://github.com/brianm/really-executable-jars-maven-plugin -->
+<build>
+  <plugins>
+    <plugin>
+      <groupId>org.skife.maven</groupId>
+      <artifactId>really-executable-jar-maven-plugin</artifactId>
+      <version>1.4.1</version>
+      <configuration>
+        <flags>-Xmx1G</flags>
+        <programFile>my-command</programFile> <3>
+      </configuration>
+      <executions>
+        <execution>
+          <phase>package</phase>
+          <goals>
+            <goal>really-executable-jar</goal>
+          </goals>
+        </execution>
+      </executions>
+    </plugin>
+  </plugins>
+</build>
+----
+<1> Define that you want to build an uber-jar. See xref:maven-tooling.adoc#uber-jar-maven[Uber-Jar Creation] for more details.
+<2> By default Quarkus will append `-runner` to the uber-jar. This would make the `really-executable-jar-maven-plugin` fail, because it cannot find the file. Therefore, we must define the uber-jar name to be equal to the project name.
+<3> Define the name of the final, executable uber-jar. This might be the name of your command.
+
+=== Executing the command
+
+Add the *my-command* file to your PATH, so you can run it directly from the command line. For instance on Linux run
+[source,text,subs=attributes+]
+----
+# update PATH. This change is only temporary for the current terminal session
+export PATH="/path/to/my-command:$PATH"
+
+my-command -n Quarkus
+# will print: Hello Quarkus
+----
+
+Remember this requires a suitable JDK version to be installed. For a single executable file that runs without installing a JDK see <<Native mode support>>. For even more packaging options check the https://picocli.info/#_packaging_your_application[official Picocli documentation].
```

Comment:
Please use an anchor for `<<Native mode support>>` or things will get easily broken if the title is changed.